plugins/kubernetes/app/models/kubernetes/job_executor.rb

```diff
@@ -0,0 +1,96 @@
+# Execute a job that can build/push Docker images
+# and write progress to the local job output.
+module Kubernetes
+  class JobExecutor
+    WAIT_FOR_JOB = 5.minutes
+    TICK = 2.seconds
+
+    attr_reader :job
+    delegate :id, to: :job
+
+    def initialize(output, job:)
+      @output = output
+      @job = job
+    end
+
+    def build!(reference, repo:, tag:, push: false, registry:)
+      job_log = ""
+      k8s_job = nil
+
+      if !valid_registry_info?(registry)
+        @output.puts "### Registry server should not be empty. Aborting..."
+        return false, job_log
+      end
+
+      contents = File.read(build_job_config_path)
+      k8s_job = Kubernetes::JobConfigFile.new(contents, build_job_config_path).job
+      k8s_job.metadata.name = "docker-build-job-#{SecureRandom.hex(16)}"
```

Comment:
Makes sense and a nice addition. Is there any guideline about the convention to attach labels to different Kubernetes resources?